pkg/executor/aggfuncs/spill_deserialize_helper.go

```diff
@@ -0,0 +1,436 @@
+		return true
+	}
+	return false
+}
+
+func (s *spillDeserializeHelper) deserializePartialResult4MaxMinJSON(dst *partialResult4MaxMinJSON) bool {
+	if s.readRowIndex < s.totalRowCnt {
+		pos := int64(0)
+		bytes := s.column.GetBytes(s.readRowIndex)
+		dst.isNull = spill.DeserializeBool(bytes, &pos)
+		dst.val = spill.DeserializeBinaryJSON(bytes, &pos)
+		s.readRowIndex++
+		return true
+	}
+	return false
+}
+
+func (s *spillDeserializeHelper) deserializePartialResult4MaxMinEnum(dst *partialResult4MaxMinEnum) bool {
+	if s.readRowIndex < s.totalRowCnt {
+		pos := int64(0)
+		bytes := s.column.GetBytes(s.readRowIndex)
+		dst.isNull = spill.DeserializeBool(bytes, &pos)
+		dst.val = spill.DeserializeEnum(bytes, &pos)
+		s.readRowIndex++
+		return true
+	}
+	return false
+}
+
+func (s *spillDeserializeHelper) deserializePartialResult4MaxMinSet(dst *partialResult4MaxMinSet) bool {
+	if s.readRowIndex < s.totalRowCnt {
+		pos := int64(0)
+		bytes := s.column.GetBytes(s.readRowIndex)
+		dst.isNull = spill.DeserializeBool(bytes, &pos)
+		dst.val = spill.DeserializeSet(bytes, &pos)
+		s.readRowIndex++
+		return true
+	}
+	return false
+}
+
+func (s *spillDeserializeHelper) deserializePartialResult4AvgDecimal(dst *partialResult4AvgDecimal) bool {
+	if s.readRowIndex < s.totalRowCnt {
+		pos := int64(0)
+		bytes := s.column.GetBytes(s.readRowIndex)
+		dst.sum = spill.DeserializeMyDecimal(bytes, &pos)
+		dst.count = spill.DeserializeInt64(bytes, &pos)
+		s.readRowIndex++
+		return true
+	}
+	return false
+}
+
+func (s *spillDeserializeHelper) deserializePartialResult4AvgFloat64(dst *partialResult4AvgFloat64) bool {
+	if s.readRowIndex < s.totalRowCnt {
+		pos := int64(0)
+		bytes := s.column.GetBytes(s.readRowIndex)
+		dst.sum = spill.DeserializeFloat64(bytes, &pos)
+		dst.count = spill.DeserializeInt64(bytes, &pos)
+		s.readRowIndex++
+		return true
+	}
+	return false
+}
+
+func (s *spillDeserializeHelper) deserializePartialResult4SumDecimal(dst *partialResult4SumDecimal) bool {
+	if s.readRowIndex < s.totalRowCnt {
+		pos := int64(0)
+		bytes := s.column.GetBytes(s.readRowIndex)
+		dst.val = spill.DeserializeMyDecimal(bytes, &pos)
+		dst.notNullRowCount = spill.DeserializeInt64(bytes, &pos)
+		s.readRowIndex++
+		return true
+	}
+	return false
+}
+
+func (s *spillDeserializeHelper) deserializePartialResult4SumFloat64(dst *partialResult4SumFloat64) bool {
+	if s.readRowIndex < s.totalRowCnt {
+		pos := int64(0)
+		bytes := s.column.GetBytes(s.readRowIndex)
+		dst.val = spill.DeserializeFloat64(bytes, &pos)
+		dst.notNullRowCount = spill.DeserializeInt64(bytes, &pos)
+		s.readRowIndex++
+		return true
+	}
+	return false
+}
+
+func (s *spillDeserializeHelper) deserializeBasePartialResult4GroupConcat(dst *basePartialResult4GroupConcat) bool {
+	if s.readRowIndex < s.totalRowCnt {
+		pos := int64(0)
+		restoredBytes := s.column.GetBytes(s.readRowIndex)
+		valsBufLen := spill.DeserializeInt(restoredBytes, &pos)
+		tmpBuffer := make([]byte, valsBufLen)
```

Comment:
How about add a new spill.DeserializeBuffer that returns `*bytes.Buffer`, so all the deserizlize details can be hidden inside that function like
```
dst.valsBuf = spill.DeserializeBuffer(restoredBytes, &pos)
```

Reply:
> How about add a new spill.DeserializeBuffer that returns `*bytes.Buffer`, so all the deserizlize details can be hidden inside that function like
> 
> ```
> dst.valsBuf = spill.DeserializeBuffer(restoredBytes, &pos)
> ```

okk